src/listensocket.cpp

```diff
@@ -86,19 +86,22 @@ ListenSocket::ListenSocket(ConfigTag* tag, const irc::sockets::sockaddrs& bind_t
 	SocketEngine::SetReuse(fd);
 	int rv = SocketEngine::Bind(this->fd, bind_to);
 	if (rv >= 0)
-		rv = SocketEngine::Listen(this->fd, ServerInstance->Config->MaxConn);
+	{
+		int maxconn = static_cast<int>(ServerInstance->Config->MaxConn);
+		rv = SocketEngine::Listen(this->fd, maxconn);
+	}
 
 	if (bind_to.family() == AF_UNIX)
 	{
 		const std::string permissionstr = tag->getString("permissions");
-		unsigned int permissions = strtoul(permissionstr.c_str(), NULL, 8);
+		unsigned long permissions = strtoul(permissionstr.c_str(), NULL, 8);
 		if (permissions && permissions <= 07777)
-			chmod(bind_to.str().c_str(), permissions);
+			chmod(bind_to.str().c_str(), static_cast<__mode_t>(permissions));
```

Comment:
What is `__mode_t`? It looks like a platform-specific extension? `man 2 chmod` refers to `mode_t` which is fine but it might need a typedef in the Windows compat header.

Reply:
Replaced with `int` which is safe thanks to the check.